### PR TITLE
Added notify the contact about the new account in the portal

### PR DESCRIPTION
**Pull request by areksolek**
Added notify the contact about the new account in the portal

### CODE REVIEW

File: install/install_schema/data.sql [hunk 1]
```diff
@@ -620,7 +620,7 @@ insert  into `u_yf_emailtemplates`(`emailtemplatesid`,`name`,`number`,`email_tem
...
...>\r\n						<td style=\"padding:0 1em 10px 0;font-family:Arial, \'Sans-serif\';font-size:13px;color:#888;white-space:nowrap;\">Dear $(params : login)$,<br />\r\n							Your account has been created successfully. Below are your username and password:<br /><br />\r\n							Your username: $(params : login)$<br />\r\n							Your password: $(params : password)$\r\n						</td>\r\n					</tr>\r\n					<tr>\r\n						<td style=\"padding:0 1em 10px 0;font-family:Arial, \'Sans-serif\';font-size:13px;color:#888;white-space:nowrap;\">\r\n							Please log in to access all of the Portal features.<br /><br />\r\n							If you have any questions or need any assistance, please send us an email to help@yetiforce.com.\r\n						</td>\r\n					</tr>\r\n				</table>\r\n			</div>\r\n		</td>\r\n	</tr>\r\n	<tr>\r\n		<td style=\"background-color:#f6f6f6;color:#888;border-top:1px solid #ccc;font-family:Arial, \'Sans-serif\';font-size:11px;\">\r\n			<div style=\"float:right;\">\r\n				$(organization : mailLogo)$\r\n			</div>\r\n			<p>\r\n				<span style=\"font-size:12px;\">\r\n					$(translate : LBL_EMAIL_TEMPLATE_FOOTER)$\r\n				</span>\r\n			</p>\r\n		</td>\r\n	</tr>\r\n</table>','YetiPortalRegister','1',NULL);
```
Comment (TomaszQr) [inline install/install_schema/data.sql]: Added information about address, and improve logo

File: languages/en-US/HelpDesk.json [hunk 2]
```diff
@@ -78,6 +78,7 @@
     "LBL_ACCOUNTS_NO_ACTIVE_SERVICE_CONTRACTS": "The Account has no active SLA",
     "LBL_SET_SERVICE_CONTRACTS_REFERENCE_DESC": "Pressing this button will relate this SLA to ticket.",
     "LBL_NOTICE_WELCOME": "Welcome to platform",
+    "LBL_NOTICE_WELCOME_TO_PORTAL": "Welcome to the YetiForce Client Portal!",
```
Comment (TomaszQr) [inline languages/en-US/HelpDesk.json]: remove

File: modules/Settings/WebserviceUsers/models/Record.php [hunk 3]
```diff
@@ -340,4 +341,34 @@ public function delete()
...
+
+	/**
+	 * Send mails with access.
+	 *
+	 * @return void
+	 */
+	public function sendEmail()
+	{
+		$moduleName = 'Contacts';
+		$recordModel = Vtiger_Record_Model::getInstanceById($this->get('crmid'), $moduleName);
```
Comment (TomaszQr) [inline modules/Settings/WebserviceUsers/models/Record.php]: if  crmid is empty ?

File: modules/Settings/WebserviceUsers/models/Record.php [hunk 4]
```diff
@@ -340,4 +341,45 @@ public function delete()
...
+
+	/**
+	 * Send mails with access.
+	 *
+	 * @return void
+	 */
+	public function sendEmail()
+	{
+		$moduleName = 'Contacts';
+		if (!empty($url = Settings_WebserviceApps_Module_Model::getServers())) {
```
Comment (TomaszQr) [inline modules/Settings/WebserviceUsers/models/Record.php]: ?

File: config/ConfigTemplates.php [hunk 5]
```diff
@@ -39,6 +39,12 @@
 			'validation' => '\App\Validator::bool',
 			'sanitization' => '\App\Purifier::bool'
 		],
+		'ENABLE_EMAIL_PORTAL' => [
```
Comment (rskrzypczak) [inline config/ConfigTemplates.php]: camelCase

File: modules/Settings/WebserviceUsers/models/Record.php [hunk 6]
```diff
@@ -340,4 +353,39 @@ public function delete()
...
+	 * @return void
+	 */
+	public function sendEmail()
+	{
+		if (empty($this->get('crmid'))) {
+			return;
+		}
+		$moduleName = 'Contacts';
+		$recordModel = Vtiger_Record_Model::getInstanceById($this->get('crmid'), $moduleName);
+		$moduleModel = Vtiger_Module_Model::getInstance($moduleName);
```
Comment (rskrzypczak) [inline modules/Settings/WebserviceUsers/models/Record.php]:
```suggestion
```

File: modules/Settings/WebserviceUsers/models/Record.php [hunk 7]
```diff
@@ -340,4 +353,39 @@ public function delete()
...
+		$emailsFields = $moduleModel->getFieldsByType('email');
```
Comment (rskrzypczak) [inline modules/Settings/WebserviceUsers/models/Record.php]: check permission to send emails (field in contact)
```suggestion
		$emailsFields = $recordModel->getModule()->getFieldsByType('email');
```

File: modules/Settings/WebserviceUsers/actions/SaveAjax.php [hunk 8]
```diff
@@ -32,10 +32,26 @@ public function process(\App\Request $request)
...
+			}
 		} else {
 			$recordModel = Settings_WebserviceUsers_Record_Model::getCleanInstance($typeApi);
+			$recordModel->setData($data);
 		}
-		$result = $recordModel->save($data);
+
+		try {
+			$recordSave = $recordModel->save();
+			if (true == \App\Config::api('enableEmailPortal')) {
```
Comment (rskrzypczak) [inline modules/Settings/WebserviceUsers/actions/SaveAjax.php]:
```suggestion
			if (\App\Config::api('enableEmailPortal')) {
```

**Pull request merged**